src/module.ts

```diff
@@ -0,0 +1,28 @@
+import type { NitroModule, NitroModuleInput } from "./types";
+
+export function defineNitroModule(def: NitroModule) {
+  return def;
+}
+
+export async function resolveNitroModule(
+  mod: NitroModuleInput
+): Promise<NitroModule> {
+  if (typeof mod === "string") {
+    // @ts-ignore
+    globalThis.defineNitroModule =
+      // @ts-ignore
+      globalThis.defineNitroModule || defineNitroModule;
+    mod = (await import(mod).then((m) => m.default)) as NitroModule;
```

Comment:
@Barbapapazes FYI, this works in nitro repo because in stup mode, jiti adds typescript support. We need explicit jiti if we wanted to support ts.